build.sbt

```diff
@@ -38,4 +37,86 @@ excludeDependencies ++= Seq(
 
 assemblyOutputPath in assembly := file("/tmp/conseil.jar")
 
-scalacOptions ++= ScalacOptions.common
\ No newline at end of file
+scalacOptions ++= ScalacOptions.common
+
+//uses git tags to generate the project version
+//see https://github.com/sbt/sbt-git#versioning-with-git
+enablePlugins(GitVersioning)
+
+/* The versioning scheme is
+ *  - use a major number as the platform version
+ *  - add a date reference in form of yyww (year + week in year)
+ *  - use the latest git tag formatted as "ci-release-<xyz>" and take the numbers from xyz, increasing it
+ * Compose the three separated by "dots" to have the version that will be released
+ * The Git plugin will add a trailing "-SNAPSHOT" if there are locally uncommitted changes
+ */
+val majorVersion = 0
+
+/* This allows to extract versions from past tags, not directly applied to
+ * the current commit
+ */
+git.useGitDescribe := true
+
+//defines how to extract the version from git tagging
+git.gitTagToVersionNumber := { tag: String =>
+  if(Versioning.releasePattern.findAllIn(tag).nonEmpty)
+    Some(Versioning.generate(major = majorVersion, date = java.time.LocalDate.now, tag = tag))
+  else
+    None
+}
+
+//custom task to create a new release tag
+lazy val prepareReleaseTag = taskKey[String]("Use the current version to define a git-tag for a new release")
+prepareReleaseTag := Versioning.prepareReleaseTagDef.value
+
+/* A command to call the "git tag" commands (from sbt-git) with custom args.
+ * Allows any automated environment (e.g. jenkins, travis) to call
+ * "sbt gitTag" when a new release has been just published, thus bumping the versioning
+ * tag and pushing to git
+ * In turn, sbt will pick it up for the new version definition
+ */
+lazy val gitTagCommand = Command.command("gitTag") { state =>
+  val extracted = Project.extract(state)
+  val (state2, tag) = extracted.runTask(prepareReleaseTag, state)
+  //we might want to check out only for non-snapshots?
+  println(s"About to tag the new release as '$tag'")
+  //we might want to read the message from the env or from a local file
```

Comment:
I'm refraining here to `push` the tag to `master` too.

It might be wiser to let the user do that manually